cpp/src/arrow/compute/api.h

```diff
@@ -21,7 +21,8 @@
 #include "arrow/compute/context.h"  // IWYU pragma: export
 #include "arrow/compute/kernel.h"   // IWYU pragma: export
 
-#include "arrow/compute/kernels/cast.h"  // IWYU pragma: export
-#include "arrow/compute/kernels/hash.h"  // IWYU pragma: export
+#include "arrow/compute/kernels/boolean.h" // IWYU pragma: export
```

Comment:
@wesm @xhochy Can we export this?

Reply:
I have no expericence with IWYU yet, so @wesm needs to answer.

Reply:
Yes, that's fine

Reply:
OK. Thanks for confirming this!